packages/gestalt/src/Callout.js

```diff
@@ -84,15 +98,27 @@ const CalloutLink = ({
       smMarginTop="auto"
       smMarginBottom="auto"
     >
-      <Button
-        accessibilityLabel={accessibilityLabel}
-        color={color}
-        href={href}
-        onClick={onClick}
-        role="link"
-        size="lg"
-        text={label}
-      />
+      {isLink && (
+        <Button
+          accessibilityLabel={accessibilityLabel}
+          color={color}
+          href={data.href}
```

Comment:
I'm getting a flow here that I'm not sure how to fix. `Cannot get data.href because property href is missing in ButtonData`. I thought adding in the `isLink` check would work, but it isn't 🤔

Reply:
Ohh I wonder if instead of adding `role` we could just make `href` optional? If given, href, assume role is a link and if not, assume it's a button.

Reply:
yeah, this is expected, you have to segment Flowtype in a particular, way I'm not sure we can accomplish that here...  i might patch and play around.

Reply:
I like the idea of using the presence of `href` instead of specifying a role **and** an href. (doing a similar thing in the new Dropdown) 
I think you could simplify even more as well:

<Button
          accessibilityLabel={accessibilityLabel}
          color={color}
          href={href ? href : undefined} //this may get linted to {href || undefined}
          onClick={onClick}
          role={href ? "link" : "button"}
          size="lg"
          text={label}
        />

Reply:
yeah, I looked at it, and this looks ok. The API here is very reduced for Button so this approach makes sense. @ayeshakmaz 's implementation is great.

Reply:
One concern, is that Links on Callout now open on the same page. I wonder if we should set it to 'blank'. This will be fixed the `useNavigation` hook. any thoughts for the interim state?

Reply:
I'll set the target to blank! I noticed it during testing and it was a really bad experience

Reply:
> I like the idea of using the presence of `href` instead of specifying a role **and** an href. (doing a similar thing in the new Dropdown)
> I think you could simplify even more as well:
> 
> ```
> <Button
>           accessibilityLabel={accessibilityLabel}
>           color={color}
>           href={href ? href : undefined} //this may get linted to {href || undefined}
>           onClick={onClick}
>           role={href ? "link" : "button"}
>           size="lg"
>           text={label}
>         />
> ```

Flow probably doesn't allow this. 

Button is segmented so Button role='button' can't have href/target props.  The only way to do this woulde {...href ? { href } : {} } but this aint good for codemods, spreading on Gestalt components makes props opaque.

Reply:
> I'll set the target to blank! I noticed it during testing and it was a really bad experience

as per discussion, leaving this change to future PR to not change more behavior and break stuff, we should see use cases and decide if we should add target prop or not or default to blank.